api/apps/api/src/modules/clone/infra/export/mark-export-as-failed.handler.ts

```diff
@@ -0,0 +1,40 @@
+import { ApiEventsService } from '@marxan-api/modules/api-events';
+import { CommandHandler, IInferredCommandHandler } from '@nestjs/cqrs';
+
+import { ResourceKind } from '@marxan/cloning/domain';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+
+import { MarkExportAsFailed } from './mark-export-as-failed.command';
+
+@CommandHandler(MarkExportAsFailed)
+export class MarkExportAsFailedHandler
+  implements IInferredCommandHandler<MarkExportAsFailed> {
+  constructor(private readonly apiEvents: ApiEventsService) {}
+
+  async execute({
+    resourceKind,
+    resourceId,
+    exportId,
+  }: MarkExportAsFailed): Promise<void> {
+    const kind =
+      resourceKind === ResourceKind.Project
+        ? API_EVENT_KINDS.project__export__failed__v1__alpha
+        : null;
+
+    if (!kind) {
+      // TODO update with Scenario once supported.
+      return;
+    }
+
+    await this.apiEvents.createIfNotExists({
+      kind,
+      topic: resourceId.value,
+      externalId: kind + exportId,
```

Comment:
can we stick to a fixed pattern for these `externalId`s for this specific functionality? with or without hyphen between `kind` and `exportId`... I'd generate these `externalId`s through a function so that we can apply the same template consistently (as long as we always use the function to generate the value of this property 😬)

Reply:
We've added `ApiEventsService.composeExternalId(id: string, kind: API_EVENT_KINDS): string` method for ensuring consistency :+1: